File: AppOptimize/app/src/main/java/com/gl/appoptimize/LeakActivity.java
package com.gl.appoptimize;

import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

public class LeakActivity extends AppCompatActivity {

    private final String TAG = "LeakActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_leak);

        // 跳转到新的LeakActivity里执行这段代码，然后退出这个LeakActivity。执行gc。
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
            }
        }, 15000);
    }

}

File: app/src/main/java/com/gl/appoptimize/memory/LeakActivity.kt
package com.gl.appoptimize.memory

import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import com.gl.appoptimize.R

class LeakActivity : AppCompatActivity() {
    private val TAG = "LeakActivity"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_leak)

        // 跳转到新的LeakActivity里执行这段代码，然后退出这个LeakActivity。执行gc。
        Handler().postDelayed({ }, 15000)
    }
}